File: src/main/java/io/myosotisdev/utopianism/entity/PlayerData.java
package io.myosotisdev.utopianism.entity;

import io.myosotisdev.utopianism.battle.BattleClass;
import io.myosotisdev.utopianism.modules.Experience;
import io.myosotisdev.utopianism.modules.expert.SkillExpert;
import io.myosotisdev.utopianism.modules.player.ImplPlayer;
import io.myosotisdev.utopianism.util.SingleSelectionModel;

import java.util.HashMap;
import java.util.Map;

public class PlayerData extends LivingEntityData
{
    private SingleSelectionModel<String, BattleClass> battleClassModel;
    private HashMap<SkillExpert, Experience> expertExperiences;

    public PlayerData(ImplPlayer player)
    {
        super(player);
    }

    public ImplPlayer getPlayer()
    {
        return ((ImplPlayer) getEntity());
    }

    public SingleSelectionModel<String, BattleClass> getBattleClassModel()
    {
        return this.battleClassModel;
    }

    public Map<SkillExpert, Experience> getExperts()
    {
        return this.expertExperiences;
    }

    public int getExpertLevel(SkillExpert expert)
    {
        return this.expertExperiences.get(expert).getLevel();
    }
}

File: src/main/java/io/myosotisdev/utopianism/entity/PlayerData.kt
package io.myosotisdev.utopianism.entity

import io.myosotisdev.utopianism.battle.BattleClass
import io.myosotisdev.utopianism.modules.Experience
import io.myosotisdev.utopianism.modules.expert.SkillExpert
import io.myosotisdev.utopianism.modules.player.MsPlayer
import io.myosotisdev.utopianism.util.SingleSelectionModel

class PlayerData(player: MsPlayer?) : LivingEntityData(player!!)
{
    val battleClassModel: SingleSelectionModel<String, BattleClass>? = SingleSelectionModel()
    private val expertExperiences: HashMap<SkillExpert, Experience> = HashMap()
    val player: MsPlayer
        get() = entity as MsPlayer
    val experts: Map<SkillExpert, Experience>
        get() = expertExperiences

    fun getExpertLevel(expert: SkillExpert): Int
    {
        return expertExperiences[expert]!!
                .level
    }
}